### ISSUE TITLE
'Make All' problem with suggested fix

### ISSUE THREAD

**Issue by TimHe95** [facebook/rocksdb#5159]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://www.facebook.com/groups/rocksdb.dev

### My Env:
* CentOS 7.3 (`uname -r`3.10.0-514.26.2.el7.x86_64)  
* gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)  

### Expected behavior
I follow the install cmd [here](https://github.com/facebook/rocksdb/blob/master/INSTALL.md). During `make all`(all the dependencies has been installed), I encountered: 

### Actual behavior
```
  ...
  CCLD     db_universal_compaction_test
  CC       tools/trace_analyzer_test.o
  CCLD     trace_analyzer_test
  CC       util/repeatable_thread_test.o
  CCLD     repeatable_thread_test
  CC       db/range_tombstone_fragmenter_test.o
  CCLD     range_tombstone_fragmenter_test
  CC       db/range_del_aggregator_test.o
  CCLD     range_del_aggregator_test
  CC       table/sst_file_reader_test.o
table/sst_file_reader_test.cc: In function ‘std::string rocksdb::EncodeAsString(uint64_t)’:
table/sst_file_reader_test.cc:22:36: error: expected ‘)’ before ‘PRIu64’
   snprintf(buf, sizeof(buf), "%08" PRIu64, v);
                                    ^~~~~~
table/sst_file_reader_test.cc:22:45: error: conversion lacks type at end of format [-Werror=format=]
   snprintf(buf, sizeof(buf), "%08" PRIu64, v);
                                             ^
table/sst_file_reader_test.cc:22:45: error: too many arguments for format [-Werror=format-extra-args]
cc1plus: all warnings being treated as errors
make: *** [table/sst_file_reader_test.o] Error 1
```

### Suggested Fix
**ADD** `#define __STDC_FORMAT_MACROS` just **BEFORE** `#include <inttypes.h>` of /rocksdb-master/table/sst_file_reader_test.cc

**Comment by riversand963** [comment]
@TimHe95 I haven't run RocksDB on the same platform as yours. Would it be possible for you to submit a PR?

**Comment by mrambacher** [comment]
Fixed by #5402
